src/main/java/core/basesyntax/service/operations/BalanceOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void getOperation(List<FruitTransaction> transactions) {
+        List<FruitTransaction> transactionsSelected = transactions.stream()
+                .filter(a -> a.getOperation() == FruitTransaction.Operation.BALANCE)
+                .toList();

```

Comment:
This stream operation is redundant here and should be removed. The filtering logic should be placed outside of the operation handler.